app/javascript/packages/document-capture/components/submission-complete.jsx

```diff
@@ -18,9 +20,19 @@ import React from 'react';
  * @param {SubmissionCompleteProps} props Props object.
  */
 function SubmissionComplete({ resource }) {
-  const response = resource.read();
+  resource.read();
 
-  return <>Finished sending: {JSON.stringify(response)}</>;
+  function submitToSelf() {
+    const form = document.querySelector('.js-document-capture-form');
+    if (form) /** @type {HTMLFormElement} */ (form).submit();
```

Comment:
1. Just to check my TS hints understanding... could we assign the type as a nullable type on the line above and then rely on the if check or even a null-safe navigator for it?

```js
const form = /** @type {HTMLFormElement?} */ document.querySelector('.js-document-capture-form');
form?.submit();
// or:
if (form) {
  form.submit();
}
```

2. Is there an eslint rule about always requiring braces for if statements? I am a big fan of "no optional braces" which I think in eslint land is [curly: all](https://eslint.org/docs/rules/curly), and also [brace-style: one true brace](https://eslint.org/docs/rules/brace-style)

Reply:
> 1. Just to check my TS hints understanding... could we assign the type as a nullable type on the line above and then rely on the if check or even a null-safe navigator for it?

Yep, that could work too. It reads a bit more expressively as well. I'll make the change 👍 

> 2. Is there an eslint rule about always requiring braces for if statements? I am a big fan of "no optional braces" which I think in eslint land is [curly: all](https://eslint.org/docs/rules/curly), and also [brace-style: one true brace](https://eslint.org/docs/rules/brace-style)

Yep. In fact, I'm much more accustomed to having this rule enforced, so very fine to have it. At least based on my history of using both the rule and Prettier, I'm assuming Prettier will not apply one way or the other, so it shouldn't interfere with ESLint.

Given your first point, however, it won't be needed. I can still make the rule change though.

Reply:
> [...] and also [brace-style: one true brace](https://eslint.org/docs/rules/brace-style)

Prettier _is_ opinionated about this. If the braces are present, they'll be formatted 1TBS, so not something we need to enforce by ESLint. But it's also [a default rule inherited from Airbnb](https://github.com/airbnb/javascript/blob/a24dc34a4a2748c99006a48e997aa0a06b1d4d94/packages/eslint-config-airbnb-base/rules/style.js#L20-L21). So we're doubly checked on this one 😄 

![image](https://user-images.githubusercontent.com/1779930/90690031-fedd7f00-e23e-11ea-85b2-62b7d7406b85.png)

Reply:
> I am a big fan of "no optional braces" which I think in eslint land is [curly: all](https://eslint.org/docs/rules/curly)

Just to be clear, this rule is specified in the default Airbnb ruleset, but as `curly: 'multi'` so, while not explicitly documented, someone had made the conscious choice not to use `curly: 'all'` 🤷  We'd be overriding that. I'm quite okay with that, but figured I'd put it out there.

https://github.com/airbnb/javascript/blob/5b462a04cdfebba712a3df18926edfcb4103d2a9/packages/eslint-config-airbnb-base/rules/best-practices.js#L25-L26

Reply:
> 1. Just to check my TS hints understanding

If it's for understanding, just a couple notes:

- With the sort of inline type annotation from your snippet, if you mean it as a type cast in the traditional sense, you'd always want to include the parentheses around the thing being cast.
   - `const form = /** @type {HTMLFormElement?} */ (document.querySelector('.js-document-capture-form'));`
   - Sorta related: https://prettier.io/blog/2020/03/21/2.0.0.html#ultimate-fix-for-closure-style-type-casts-7791httpsgithubcomprettierprettierpull7791-by-thorn0httpsgithubcomthorn0-7011httpsgithubcomprettierprettierpull7011-by-evilebottnawihttpsgithubcomevilebottnawi
- But we don't need to cast it this way. We can use the more traditional JSDoc annotation of a variable instead.

```js
/** @type {HTMLFormElement?} */
const form = document.querySelector('.js-document-capture-form');
```

The main difference is that in the latter example, if there's no possible way that the variable could be of that type, TypeScript would alert us:

![image](https://user-images.githubusercontent.com/1779930/90690911-7c55bf00-e240-11ea-9380-0255bf2a1503.png)

Whereas, if we type-cast, TypeScript will just... take our word for it 😄 

![image](https://user-images.githubusercontent.com/1779930/90690855-6516d180-e240-11ea-9649-8d40612b18ed.png)

Reply:
Updates to use optional chaining and type-annotate `form` as nullable in c018441.

The `curly` rule would have a pretty broad impact to other files for preexisting issues. I'll open a separate pull request for it since it's not strictly needed here.

Reply:
Ah perfect, I didn't realize the line-above syntax for `@type` was so useful like that. Thanks!

Reply:
> The `curly` rule would have a pretty broad impact to other files for preexisting issues. I'll open a separate pull request for it since it's not strictly needed here.

Opened at #4094.